WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -282,12 +283,20 @@ bool TryRemoveTrackedAutoStart(IWallet wallet)
 		await WaitAndHandleResultOfTasksAsync(nameof(trackedAutoStarts), trackedAutoStarts.Values.Select(x => x.Task).ToArray()).ConfigureAwait(false);
 	}
 
-	private void ScheduleRestartAutomatically(IWallet walletToStart, ConcurrentDictionary<IWallet, TrackedAutoStart> trackedAutoStarts, bool stopWhenAllMixed, bool overridePlebStop, CancellationToken stoppingToken)
+	private void ScheduleRestartAutomatically(IWallet walletToStart, ConcurrentDictionary<IWallet, TrackedAutoStart> trackedAutoStarts, bool stopWhenAllMixed, bool overridePlebStop, bool wasRestartedAutomatically, CancellationToken stoppingToken)
 	{
+		var delayRestart = 30;
 		if (trackedAutoStarts.ContainsKey(walletToStart))
 		{
-			walletToStart.LogDebug($"AutoStart was already scheduled.");
-			return;
+			walletToStart.LogDebug($"AutoStart was already scheduled, cancel the last task and do not wait.");
+			trackedAutoStarts[walletToStart].CancellationTokenSource.Cancel();
+			trackedAutoStarts.Remove(walletToStart, out _);
+			delayRestart = 0;
+		}
+
+		if (!wasRestartedAutomatically)
+		{
+			NotifyWalletStartedCoinJoin(walletToStart);
```

Comment:
Try to not add extra parameters to the business logic only used for notification purposes. The state machine should be able to handle this in the ViewModel.